syntax = "proto3";

package catalog;

import "google/protobuf/empty.proto";

service AuthorController {
    rpc List(AuthorListRequest) returns (stream Author) {}
    rpc Create(Author) returns (Author) {}
    rpc Retrieve(AuthorRetrieveRequest) returns (Author) {}
    rpc Update(Author) returns (Author) {}
    rpc Destroy(Author) returns (google.protobuf.Empty) {}
}

message Author {
    int32 id = 1;
    string first_name = 2;
    string last_name = 3;
    string date_of_birth = 4;
    string date_of_death = 5;
}

message AuthorListRequest {
}

message AuthorRetrieveRequest {
    int32 id = 1;
}